Realm/RLMRealm.mm

```diff
@@ -382,6 +402,17 @@ + (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration er
                 }
 
                 RLMRealmSetSchemaAndAlign(realm, schema);
+            } catch (SchemaMismatchException const& exception) {
+                if (configuration.deleteRealmIfMigrationNeeded) {
+                    BOOL success = [[NSFileManager defaultManager] removeItemAtURL:configuration.fileURL error:nil];
+                    if (success) {
+                        realm = nil;
+                        return [self realmWithConfiguration:configuration error:error];
```

Comment:
This code path isn't tested and worries me a bit. I suspect it should be invalidating the Realm if `beganReadTransaction` was set.

Could you add a unit test and determine if the realm should be invalidated before set to `nil`?